### DISCUSSION TITLE
Running StereoKit on Blazor

### DISCUSSION THREAD

**Discussion by ddobric** [StereoKit/StereoKit#465] [Ideas]
Any intention to run the StereoKit in Blazor?

Right now, when the code is run inside blazor page, the following exception is thrown:
```

System.DllNotFoundException: StereoKitC
   at StereoKit.SK.InitializeCall(SKSettings settings)
   at StereoKit.SK.Initialize(SKSettings settings)
```

The initialization code looks like:
```

            SKSettings settings = new SKSettings
            {
                appName = "StereoKitBlazor",
                assetsFolder = "Assets",
            };

            if (!SK.Initialize(settings))
                Environment.Exit(1);
```

Thanks
Damir

**Comment by maluoi** [comment]
Yeah! This is a big chunk of the work I've been doing recently. It's not likely to go into this next release though, it has been a lot of work! Blazor's PInvoke seems to behave very differently from how it does in normal C# (particularly with callbacks!), so I'm still working through a way to get that behaving :)

StereoKit's Simulator does work on web from native code already, I also still need to add a WebXR backend.

**Comment by ddobric** [comment]
I'm happy you are working on that. Blazor is a very interesting and important scenario. 
It would be nice to provide a few more details about this. Where exactly do you use P/Invoke, what is exactly different?
